jdaviz/configs/imviz/plugins/links_control/links_control.py

```diff
@@ -52,8 +60,20 @@ def _on_link_updated(self, msg):
         self.wcs_use_fallback = msg.wcs_use_fallback
         self.wcs_use_affine = msg.wcs_use_affine
 
+    def _on_new_app_data(self, msg):
+        if self.app._jdaviz_helper._in_batch_load > 0:
+            return
+        components = [str(comp) for comp in msg.data.main_components]
+        if "ra" in components or "Lon" in components:
+            # linking currently removes any markers, so we want to skip
+            # linking immediately after new markers are added (see imviz.helper.link_image_data).
+            # Eventually we'll probably want to support linking WITH markers, at which point this
+            # if-statement should be removed.
+            return

```

Comment:
this is a temporary workaround to mimic the previous behavior where markers handle linking themselves within add_markers and are *removed* when future calls are made to linking.

Reply:
@pllim - if you have suggestions for a better way to detect if `msg.data` (which should be the new data collection entry object) is a markers entry as opposed to an image, let me know!

Reply:
This is probably ok for now, especially given we'll revisit this in the future.